Rtp/RtpClient.Fields.cs

```diff
@@ -59,7 +59,7 @@ public partial class RtpClient
         //Collection to handle the dispatch of events.
         //Notes that Collections.Concurrent.Queue may be better suited for this in production until the ConcurrentLinkedQueue has been thoroughly engineered and tested.
         //The context, the item, final, recieved
-        readonly Media.Common.Collections.Generic.ConcurrentLinkedQueueSlim<System.Tuple<RtpClient.TransportContext, Common.BaseDisposable, bool, bool>> m_EventData = new Media.Common.Collections.Generic.ConcurrentLinkedQueueSlim<System.Tuple<RtpClient.TransportContext, Common.BaseDisposable, bool, bool>>();
+        readonly Media.Common.Collections.Generic.ConcurrentLinkedQueueSlim<(RtpClient.TransportContext Context, Common.BaseDisposable Packet, bool Final)> m_EventData = new Media.Common.Collections.Generic.ConcurrentLinkedQueueSlim<(RtpClient.TransportContext Context, Common.BaseDisposable Packet, bool Final)>();
```

Comment:
Your missing the bool for sent / received, we can change this but it would be harder. Let me know if you want my help refactoring this further

Reply:
You can just include another bool here, ```bool Sent``` and use that in the logic below.

Reply:
Well, it was set to `true` in every call, and only received branch was taken.

Should I return old variant?

Reply:
I would have to check into that it was always true to confirm but as for the changes I would suggest to make Sent it's own property of the Tuple...

```
readonly Media.Common.Collections.Generic.ConcurrentLinkedQueueSlim<(RtpClient.TransportContext Context, Common.BaseDisposable Packet, bool Final)> m_EventData = new Media.Common.Collections.Generic.ConcurrentLinkedQueueSlim<(RtpClient.TransportContext Context, Common.BaseDisposable Packet, bool Final, bool SentPacket)>();
```